app/src/main/java/org/p2p/wallet/root/DecorSystemBarsDelegate.kt

```diff
@@ -0,0 +1,68 @@
+package org.p2p.wallet.root
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.WindowCompat
+import androidx.core.view.WindowInsetsControllerCompat
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import android.graphics.Color
+import android.os.Build
+
+enum class SystemIconsStyle {
+    BLACK, WHITE
+}
+
+class DecorSystemBarsDelegate constructor(
+    private val activity: AppCompatActivity
+) : DefaultLifecycleObserver {
+
+    private var navigationBarStyle: SystemIconsStyle = SystemIconsStyle.BLACK
+    private var statusBarStyle: SystemIconsStyle = SystemIconsStyle.BLACK
+
+    private val windowInsetsController: WindowInsetsControllerCompat
+        get() = WindowCompat.getInsetsController(activity.window, activity.window.decorView)
+
+    init {
+        activity.lifecycle.addObserver(this)
+    }
+
+    override fun onCreate(owner: LifecycleOwner) {
+        setupWindowInsets()
+        setDefaultSystemBarStyle()
+    }
+
+    fun setDefaultSystemBarStyle() {
+        setStatusBarStyle(statusBarStyle)
+        setNavigationBarStyle(navigationBarStyle)
+    }
+
+    fun setSystemBarStyle(style: SystemIconsStyle) {
+        setStatusBarStyle(style)
+        setNavigationBarStyle(style)
+    }
+
+    fun setStatusBarStyle(style: SystemIconsStyle) {
+        windowInsetsController.isAppearanceLightStatusBars = when (style) {
+            SystemIconsStyle.BLACK -> true
+            SystemIconsStyle.WHITE -> false
+        }
+    }
+
+    fun setNavigationBarStyle(style: SystemIconsStyle) {
+        windowInsetsController.isAppearanceLightNavigationBars = when (style) {
```

Comment:
windowInsetsController.isAppearanceLightNavigationBars = style is SystemIconsStyle.BLACK